### ISSUE TITLE
NumberFormatException updating buttons inside facet header of a data table

### ISSUE THREAD

**Issue by HKM-CIA** [eclipse-ee4j/mojarra#5153]
**Describe the bug**

updating buttons inside facet header of primefaces p:columns component
leads to a NumberFormatException using Mojarra, works with MyFaces

This problem was first reported to the Primefaces-Team [#9245](https://github.com/primefaces/primefaces/issues/9245)

**To Reproduce**

Please use the attached primefaces test project to reproduce the beahviour.

1. mvn clean jetty:run -Pmyfaces23 - WORKS!
2. mvn clean jetty:run -Pmojarra23 - FAILS.

**Expected behavior**

The buttons should switch their state between disabled and enabled.

[primefaces-test.zip](https://github.com/eclipse-ee4j/mojarra/files/9731434/primefaces-test.zip)

**Comment by BalusC** [comment]
Reproducer was unusable because PrimeFaces 11.0.8 is in a private repo. I've used 11.0.0 instead which is available in central. I was able to reproduce the problem.

Problem boils down to that client ID row index parser used by standard UIData implementation totally didn't expect an extra naming container ID between the UIData component and the target component because the UIData renderer doesn't support any other child component than UIColumn, and the UIColumn by itself doesn't prepend its ID to the client ID.

So, given a normal usage:
```
<h:form id="form">
    <h:dataTable id="table">
        <h:column id="column">
            <f:facet name="header">
                <h:commandButton id="button">
```
the generated client ID of the button will become `form:table:0:button`. The UIData implementation expects that the row index is right after own client ID of `form:table`.

However, in your test case, a `<p:columns>` is being used. This apparently prepends its ID to the client ID:
```
<h:form id="form">
    <p:dataTable id="table">
        <p:columns id="columns" value="#{...}" var="...">
            <f:facet name="header">
                <h:commandButton id="button">
```
the generated client ID of the button will become `form:table:columns:0:button`. The UIData implementation totally didn't expect that and blindly assumes that `columns` is the row index.
